packages/nimble/R/BUGS_modelDef.R

```diff
@@ -2715,6 +2718,39 @@ modelDefClass$methods(genVarNames = function() {
     varNames <<- c(names(varInfo), names(logProbVarInfo))
 })
 
+modelDefClass$methods(warnRHSonlyDynIdx = function() {
+    ## Warn if dynamic indexing involves non-constant RHS-only nodes as this causes
+    ## additional dependencies and slower computations.
+    for(i in seq_along(declInfo)) {
+        decl <- declInfo[[i]]
+        if(exists('dynamicIndexInfo', decl) && length(decl[['dynamicIndexInfo']])) {
+            ## Determine vars used in dynamic indexing.
+            vars <- lapply(decl[['dynamicIndexInfo']], function(x) {
+                if(exists('indexCode', x))
+                    return(all.vars(x[['indexCode']]))
+            })
+            vars <- unique(unlist(vars))
+            vars <- vars[vars %in% decl$rhsVars]
+            ## Evaluate indexing to determine nodes used in dynamic indexing.
+            nr <- min(50, nrow(decl$unrolledIndicesMatrix))  # avoid doing full expansion for speed
+            nodes <- lapply(seq_along(vars), function(idx) {
```

Comment:
This approach is extensive.  I guess it makes sense.  Another idea would be to issue the warning simply if an index variable *ever* appears as a LHS node.  I think the difference is that the approach here more thoroughly checks for the first (up to) 50 instances of the index variable, so it tries to handle a case where some indices of the variable are RHSonly and some are not.  Is that the idea?

Would it make sense to wrap all of this in a `try` so that if the checking itself has any glitches it is not an obstacle to the user?  Separately, would it make sense to put this entire checking step behind an option flag, with default being on.  That way, if a user reports a bug in the checking itself, we can simply tell them to set an option FALSE and they will be able to move ahead.  That would seem easier than providing a branch for them etc.

Reply:
Yes, idea was as usual that we don't know how a user might split up a variable for multiple purposes. 

Not quite sure what you mean by `imply if an index variable ever appears as a LHS node`. Do you mean just to check if the index variable coincides with any variables corresponding to RHSonly nodes?

How strongly do you feel about `try`? As I've just modified it, if something goes wrong we can tell user(s) to set the option `checkRHSonlyDynamicIndex` to `FALSE` to prevent the checking from happening.

Reply:
What I meant was if we see `x[ k[i] ]`, so `k` is the index variable, then if `k` *never* appears on the LHS, it must only be RHSonly.  But I guess it is moot to think of other approaches, and in this case also more coarse.

Sure, I guess putting it all behind and option flag gives a sufficient failsafe.